ludwig/data/preprocessing.py

```diff
@@ -1211,6 +1211,15 @@ def build_dataset(
             logger.debug(f"sample {sample_ratio} of data")
             dataset_df = dataset_df.sample(frac=sample_ratio, random_state=random_seed)
 
+        sample_cap = global_preprocessing_parameters["sample_cap"]
```

Comment:
Could you refactor this section out into a separate function?

```python
dataset_df = get_sampled_dataset_df(dataset_df, sample_ratio, sample_cap)
```

Reply:
Yep. Done!